src/main/java/com/kodilla/good/patterns/food2Door/HealthyShop.java

```diff
@@ -23,11 +24,17 @@ public HealthyShop(Map<Integer, Product> productsMap) {
 
     @Override
     public boolean process(OrderRequest orderRequest) throws ProductNotFoundException {
-        for (int i = 0; i < productsMap.size(); i++) {
-            if (orderRequest.getProduct().getProductName().equals(productsMap.get(i).getProductName())) {
-                System.out.println("Order in HealthyShop has been placed.");
-                return true;
-            } else throw new ProductNotFoundException("Product " + orderRequest.getProduct() + " was not found.");
-        } return false;
+        Optional<Product> productOptional = productsMap.entrySet().stream()
+                .flatMap()
+//                .flatMap(productInMap -> productInMap.getValue())
+//                .filter(productInList -> productInList.getProductName().equals(orderRequest.getProduct().getProductName()))
+                .findAny();
+
+        if (productOptional.isPresent()) {
+            System.out.println("Order in HealthyShop has been placed.");
+            return true;
+        }
+        throw new ProductNotFoundException("Product " + orderRequest.getProduct().getProductName() + " was not found.");
```

Comment:
we wszystkich 3 shop'ach jest praktycznie taka sama mechanika. Warto albo zmodyfikowac ja tak, aby kazdy shop dzialal nieco inaczej, albo wyciagnac te mechanike do klasy abstrakcyjnej